api/src/main/java/org/apache/iceberg/Metrics.java

```diff
@@ -59,26 +62,56 @@ public Metrics(Long rowCount,
     this.upperBounds = upperBounds;
   }
 
+  /**
+   * Get the number of records (rows) in file.
+   *
+   * @return
+   */
   public Long recordCount() {
     return rowCount;
   }
 
+  /**
+   * Get the number of bytes for all fields in a file.
+   *
+   * @return a Map of fieldId to the size in bytes
+   */
   public Map<Integer, Long> columnSizes() {
     return columnSizes;
   }
 
+  /**
+   * Get the number of all values, including null and repeated.
+   *
+   * @return a Map of fieldId to the number of all values
+   */
   public Map<Integer, Long> valueCounts() {
     return valueCounts;
   }
 
+  /**
+   * Get the number of null values for all fields in a file.
+   *
+   * @return a Map of fieldId to the number of nulls
+   */
   public Map<Integer, Long> nullValueCounts() {
     return nullValueCounts;
   }
 
+  /**
+   * Get the non-null lower bound values for all fields in a file.
+   *
+   * @return a Map of fieldId to the lower bound value as a ByteBuffer
```

Comment:
Can we also add a reference to the conversion back to a value? And maybe one to the spec for that conversion?

Reply:
Done.